ui/components/component-library/select-wrapper/select-wrapper.tsx

```diff
@@ -0,0 +1,109 @@
+import React, { useState } from 'react';
+import classnames from 'classnames';
+import type { PolymorphicRef, BoxProps } from '../box';
+import { Box, Popover, PopoverPosition } from '..';
+import {
+  SelectWrapperComponent,
+  SelectWrapperProps,
+} from './select-wrapper.types';
+import { SelectContext } from './select-wrapper.context';
+
+export const SelectWrapper: SelectWrapperComponent = React.forwardRef(
+  <C extends React.ElementType = 'div'>(
+    {
+      className = '',
+      defaultValue,
+      value,
+      onValueChange,
+      placeholder,
+      isDanger,
+      isDisabled,
+      isOpen,
+      onOpenChange,
+      isMultiSelect, // Prevents the uncontrolled open state from being toggled
+      triggerComponent,
+      popoverProps,
+      children,
+      onBlur, // Controlled onBlur prop
+      ...props
+    }: SelectWrapperProps<C>,
+    ref?: PolymorphicRef<C>,
+  ) => {
+    const [uncontrolledValue, setUncontrolledValue] = useState<any | null>();
+    const [isUncontrolledOpen, setIsUncontrolledOpen] =
+      useState<boolean>(false);
+    const [referenceElement, setReferenceElement] =
+      useState<HTMLElement | null>();
+
+    const setBoxRef = (popoverRef: HTMLElement | null) => {
+      setReferenceElement(popoverRef);
+    };
+
+    const toggleUncontrolledOpen = () => {
+      setIsUncontrolledOpen(!isUncontrolledOpen);
+    };
+
+    const handleBlur = (e: any) => {
+      const wrapper = e.currentTarget;
+      const { relatedTarget } = e;
+
+      if (!wrapper.contains(relatedTarget)) {
+        // Close the popover only if the related target is not inside the wrapper
+        setIsUncontrolledOpen(false);
+        // If you have a controlled isOpen state, update it to close
+        if (onOpenChange) {
+          onOpenChange(false);
+        }
```

Comment:
nit: match shorthand notation of onBlur
```suggestion
          onOpenChange?.(false);
```

Reply:
Either works! I just think we should be consistent if we can be